addons/account/models/account_move.py

```diff
@@ -1235,14 +1235,16 @@ def _compute_payment_term_details(self):
             if invoice.show_payment_term_details:
                 sign = 1 if invoice.is_inbound(include_receipts=True) else -1
                 payment_term_details = []
-                for line in invoice.line_ids.filtered(lambda l: l.display_type == 'payment_term'):
+                for line in invoice.line_ids.filtered(lambda l: l.display_type == 'payment_term').sorted('date_maturity'):
                     payment_term_details.append({
-                        'date': format_date(self.env, line.date_maturity),
+                        'date': line.date_maturity,
                         'amount': sign * line.amount_currency,
                         'discount_date': format_date(self.env, line.discount_date),
                         'discount_amount_currency': sign * line.discount_amount_currency,
                     })
-                invoice.payment_term_details = sorted(payment_term_details, key=lambda t: t.get('date'))
+                invoice.payment_term_details = payment_term_details
+                for pay_term in invoice.payment_term_details:
+                    pay_term['date'] = format_date(self.env, pay_term.get('date'))
```

Comment:
But you don't need this diff now, that's the point :sweat_smile: 
You can format directly in the first loop

Reply:
Yeah, went a bit too fast on this one :sweat_smile: 
I'll change that in a bit